src/user/views/user_views.py

```diff
@@ -72,6 +78,47 @@ def get_serializer_class(self):
     def get_serializer_context(self):
         return {"get_subscribed": True, "get_balance": True, "user": self.request.user}
 
+    @action(detail=False, methods=["GET"], permission_classes=[IsAuthenticated])
+    def get_referred_users(self, request):
+        invited = list(
+            map(
+                lambda invited_user: {
+                    "user": DynamicUserSerializer(
+                        invited_user,
+                        _include_fields=[
+                            "author_profile",
+                            "created_date",
+                            "id",
+                        ],
+                        many=False,
+                        context={},
+                    ).data,
+                    "created_date": invited_user.created_date,
+                },
+                self.queryset.filter(invited_by=request.user.id),
+            )
+        )
+
+        earned_distributions = list(
+            Distribution.objects.filter(
+                distribution_type=REFERRAL_PROGRAM["REFERER_DISTRIBUTION_TYPE"],
+                giver_id__in=list(map(lambda invited: invited["user"]["id"], invited)),
+            )
+            .values("recipient_id", "giver_id")
+            .annotate(rsc_earned=Sum("amount"))
+        )
+
+        for invited_user in invited:
+            for dist in earned_distributions:
+                if dist["giver_id"] == invited_user["user"]["id"]:
+                    invited_user["rsc_earned"] = dist["rsc_earned"]
+
+            invited_user["benefits_expire_on"] = invited_user[
+                "created_date"
+            ] + relativedelta(months=+6)
+
+        return Response(invited)
```

Comment:
```suggestion
    def get_referred_users(self, request):
        invited = User.objects.all().filter(
            invited_by__id=request.user.id
        ).annotate(
            rsc_earned=Sum('reputation_records'), 
            benefits_expire_on=models.ExpressionWrapper(models.F('created_date') + datetime.timedelta(days=30), 
            output_field=models.DateTimeField())
    )
```
Something like this works better. You'll have to modify the serializer and logic a bit, but it is better than iterating through the queryset. You can also add a filter on the `Sum` function to filter by specific distributions.